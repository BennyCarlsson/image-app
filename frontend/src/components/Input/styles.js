import styled from 'styled-components'

export const InputWrapper = styled.div`
  display: flex;
  flex-direction: column;
`

export const StyledInput = styled.input`
  border-color: transparent;
  border-bottom: 1px solid #e9e9e9;
  outline-style: none;
  width: 320px;
  margin-bottom: 24px;
  font-weight: 400;
  font-size: 15px;
  letter-spacing: -0.25px;
  padding: 10px 0px;
  border-radius: 1px;
  &:focus {
    border-bottom: 1px solid #9b9b9b;
  }
`
export const Label = styled.label`
  margin-left: 0;
  padding-left: 0;
  color: #607bff;
  font-size: 13px;
  letter-spacing: -0.35px;
  font-weight: bold;
`
