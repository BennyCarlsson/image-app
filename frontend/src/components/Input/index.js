import React from 'react'
import { InputWrapper, Label, StyledInput } from './styles'

const Input = ({ onChange, value }) => {
  return (
    <InputWrapper>
      <Label htmlFor='filename'>Image name</Label>
      <StyledInput
        name='filename'
        id='filename'
        placeholder='Enter name...'
        type='text'
        onChange={onChange}
        value={value}
      />
    </InputWrapper>
  )
}

export default Input
