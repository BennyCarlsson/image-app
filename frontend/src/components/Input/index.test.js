import React from 'react'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import Input from '.'

describe('<Input />', () => {
  const value = 'value text'
  const mockOnChange = jest.fn()

  it('should match snapshot', () => {
    const { asFragment } = render(
      <Input onChange={mockOnChange} value={value} />
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render without crashing', () => {
    render(<Input onChange={mockOnChange} value={value} />)
    expect(screen).toBeTruthy()
  })

  it('should render input with value', () => {
    render(<Input onChange={mockOnChange} value={value} />)
    expect(screen.getByRole('textbox')).toHaveValue(value)
  })

  it('should call onChange', () => {
    render(<Input onChange={mockOnChange} value={value} />)
    expect(mockOnChange).not.toHaveBeenCalled()
    userEvent.type(screen.getByRole('textbox'), 'Hello World')
    expect(mockOnChange).toHaveBeenCalledTimes(11)
  })
})
