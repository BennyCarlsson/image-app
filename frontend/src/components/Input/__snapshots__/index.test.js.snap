// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL hjGchb"
  >
    <label
      class="sc-dkPtRN hJOcUV"
      for="filename"
    >
      Image name
    </label>
    <input
      class="sc-gsDKAQ kydZcA"
      id="filename"
      name="filename"
      placeholder="Enter name..."
      type="text"
      value="value text"
    />
  </div>
</DocumentFragment>
`;
